Core/CharacterHandler: Allow alliedrace creating.

Varjgard opened the pull request:
**Changes proposed:**
-  Allow player to create alliedrace character, 
   if player has the required achievement, 
   or if achievement requirement is set to 0 in db


**Target branch(es):** 3.3.5/master
- [ ] 3.3.5
- [x ] master


**Tests performed:** (Does it build, tested in-game, etc.)
- it builds, and player can create alliedrace characters.

Commit:
Core/CharacterHandler: Allow alliedrace creating.

Allow player to create alliedrace character, if player has the required achievement, or if achievemnet requirement is set to 0 in db

Varjgard commented:
this commit is actually passing circleci, only the fork i have is not passing it. only 1 line changed on this commit so circleci can be ignored.

Shauren commented:
No, it cannot be ignored. Class `WorldSession` does not have a function called `HasAchievement`

Varjgard commented:
`raceUnlock.HasAchievement = requirement.second.AchievementId == 0 /*|| HasAchievement(requirement.second.AchievementId)*/;`

Commit:
Core/CharacterHandler: Allow alliedraces.

Allow player to create alliedraces if achievement requirement is set to 0 in db.
